root/edit/details/AddArea.js

```diff
@@ -0,0 +1,122 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import React from 'react';
+
+import {addColon, l} from '../../static/scripts/common/i18n';
+import {lp_attributes} from '../../static/scripts/common/i18n/attributes';
+import commaOnlyList from '../../static/scripts/common/i18n/commaOnlyList';
+import formatDate from '../../static/scripts/common/utility/formatDate';
+import isDateEmpty from '../../static/scripts/common/utility/isDateEmpty';
+import DescriptiveLink from '../../static/scripts/common/components/DescriptiveLink';
+import yesNo from '../../static/scripts/common/utility/yesNo';
+
+type AddAreaEditT = {|
+  ...EditT,
+  +display_data: {|
+    ...CommentRoleT,
```

Comment:
Nitpick, though it wouldn't have an effect on the current code, `CommentRoleT` has a slightly different type. The type here should be `+comment: string | null`.

Reply:
Changed. Ideally at some point we might try to have more consistent types (and actual serializations, obvs) for stuff that should really be the same though :/